backend/src/main/kotlin/com/pk/zbtz/zbtzbackend/controllers/people/service/PostgresPeopleService.kt

```diff
@@ -0,0 +1,107 @@
+import com.pk.zbtz.zbtzbackend.utils.execution_timer.ExecutionTimer
+import org.springframework.stereotype.Service
+import kotlin.math.ceil
+
+@Service
+class PostgresPeopleService(
+    private val service: HumanService,
+    private val executionTimer: ExecutionTimer,
+
+    ): PeopleService {
+    override fun getAll(
+        nameToSearch: String?,
+        pageSize: Int?,
+        offset: Int?,
+    ): ResponseWithStatistics<GetPeopleResponse> {
+        val offset = offset ?: DEFAULT_OFFSET
+        val pageSize = pageSize ?: DEFAULT_PAGE_SIZE
+        val elapsedTimeResult = executionTimer.measure {
+            if(nameToSearch.isNullOrEmpty()){
+                service.getAll(offset).map { it.toHumanSummary() }
+            } else {
+                service.getAllByFirstNameOrSecondName(nameToSearch, offset).map { it.toHumanSummary() }
+            }
+        }
+
+        val peoplePage = elapsedTimeResult.blockResult
+
+        val statistics = getStatistics(elapsedTimeResult)
+        val limitedResult = limitResult(peoplePage, pageSize)
+        val response = GetPeopleResponse(
+            people = limitedResult,
+            nextOffset = offset + limitedResult.size,
+            totalPages = calculateTotalPages(peoplePage, pageSize),
+            totalRecords = peoplePage.size
+        )
+        return ResponseWithStatistics(
+            data = response,
+            statistics = statistics
+        )
+    }
+
+    override fun get(humanId: String): ResponseWithStatistics<Human> {
+        val elapsedTimeResult = executionTimer.measure {
+            service.getById(humanId.toInt())?.toHuman()
+        }
+        val human = elapsedTimeResult.blockResult
+        return ResponseWithStatistics(
+            data = human,
+            statistics = getStatistics(elapsedTimeResult)
+        )
+    }
+
+    override fun add(request: AddHumanRequest): ResponseWithStatistics<Human> {
+        val elapsedTimeResult = executionTimer.measure {
+            service.save(request.toPostgresHumanModel())
+        }
+
+        val human = elapsedTimeResult.blockResult.toHuman()
+        return ResponseWithStatistics(
+            data = human,
+            statistics = getStatistics(elapsedTimeResult)
+        )
+    }
+
+    override fun delete(humanId: String): ResponseWithStatistics<Unit> {
+        val elapsedTimeResult = executionTimer.measure {
+            service.delete(humanId.toInt())
+        }
+
+        return ResponseWithStatistics(
+            statistics = getStatistics(elapsedTimeResult)
+        )
+    }
+
+    private fun <T> getStatistics(
+        elapsedTimeResult: ExecutionTimer.ElapsedTimeResult<T>
+    ): Statistics =
+        Statistics(
+            accessTime = elapsedTimeResult.time,
+            //TODO get db size
+            databaseMemorySize = 1000.0,
+        )
+
+    private fun limitResult(people: List<HumanSummary>, limit: Int): List<HumanSummary> {
+        return people.take(limit)
+    }
```

Comment:
For a large database, I don't think it's a good idea to always get all records. If you change to `PageRequest` as I suggested, this problem will be solved

Reply:
I will check